### DISCUSSION TITLE
prometheus metrics which gets concourse build duration along with the build status

### DISCUSSION THREAD

**Discussion by rohithmn3** [concourse/concourse#5671] [General]
I was initially using Influxdb as a datasource and could able to fetch the concourse build duration metrics with build_status using the measurement "build finished" (FYI.. https://mockersf.github.io/docs/metrics.html)

Now I am trying to fetch similar details using Prometheus datasource. But I couldn't able to find the suitable metrics for the same - here: https://concourse-ci.org/metrics.html#whats-emitted; I couldn't able to find any metric which gives me build_duration along with build_status.

Could someone please help me here in achieving this..!

**Comment by rohithmn3** [comment]
I tried using the below query:
concourse_builds_finished{pipeline="pipeline_name",exported_job="job_name"}

But this gives me the Count of builds finished across various dimensions I guess. Not the duration of the builds.

From the code, I could see it is emitted as a histogram: https://github.com/concourse/concourse/blob/a4c9e8ed8205218b1fdd8646f5a107a6fbbb670e/atc/metric/emitter/prometheus.go#L575

But I am not able to query it. Please let me know what I am missing.

**Reply by jamieklassen** [reply]
I haven't looked very closely, but something stands out to me here: you said your promQL involved the `concourse_builds_finished` metric, but the histogram you mentioned (`buildDurationsVec`) should be called `concourse_builds_duration_seconds`:

https://github.com/concourse/concourse/blob/a4c9e8ed8205218b1fdd8646f5a107a6fbbb670e/atc/metric/emitter/prometheus.go#L233-L243

**Reply by jamieklassen** [reply]
I was able to see the buckets with `concourse_builds_duration_seconds_bucket{team="main",job="concourse"}`

**Reply by rohithmn3** [reply]
@jamieklassen yes I have seen this later in the Prometheus dashboard. Earlier I was referring the documentation https://concourse-ci.org/metrics.html#whats-emitted; so got a bit confused. Thanks for the clarification.

Now I am trying to basically merge these two metrics: `concourse_builds_finished` and `concourse_builds_duration_seconds_sum` to get the value as the duration of each job with attributes/labels as a team, pipeline, job, and a status. 

Could you please help me in achieving this. Because I am not able to find the UNIQUE label which helps in one-to-one mapping of these two metrics.

Currently trying with the below query, but really not sure what to put in ON.
`concourse_builds_duration_seconds_sum{team="team1",pipeline="pipeline1"} * on(??????????) group_left(status) concourse_builds_finished{}`

**Reply by rohithmn3** [reply]
@jamieklassen - any help you could able to do here..!

**Comment by rohithmn3** [comment]
Hi Team,

Please let me know whether it is possible to get the build_duration as a value and team, pipeline, job, and status as labels...!?
I am really finding it hard to achieve it. Could someone please help me here.

**Reply by prime89-hash** [reply]
Hi @rohithmn3 , 
Did you manage to resolve your issue, actually we also want to get concourse build duration along with the build status.
It will be really great if you could help us on the same.

**Reply by username-is-already-taken2** [reply]
Hi there, I’m just getting started with Prometheus and I was looking to create the same visualisation. It would be great if we could share what we learn?.   Eventually I would like to create a grafana dashboard and share this within the community
